torch/csrc/lazy/core/lazy_graph_executor.cpp

```diff
@@ -936,6 +930,14 @@ std::shared_ptr<LazyGraphExecutor::Async> LazyGraphExecutor::
         std::vector<BackendDataPtr> parameters_data,
         std::vector<BackendDataPtr> tensors_data,
         ComputationCache::TypePtr cached_computation) {
+  VLOG(4) << "Waiting on device barrier for device " << coll->device << " ...";
+  {
+    TORCH_LAZY_TIMED("DeviceLockWait");
+    coll->unlocker =
+        DeviceLockerArena::Get()->LockDevices({coll->device });
+  }
+  VLOG(4) << "Waiting on device barrier for device " << coll->device << " done!";
+
```

Comment:
would it be better to move it after
```
    if (noop_execution_mode_) {
      return;
    }
```
? For the threadpool implementation we only need to block the child thread instead of the main(tracing) thread

Reply:
ehh, OK. it is a bit annoying to deal with `coll` inside this syncfn, I think we can keep it like this for now.

```
execution 1
execution 2
execution 3
```

The only case that moving it inside the syncfn would help is when we want to schedule execution 2, execution 1 has not finished yet. In this case I don't think start tracing execution 3 will help the overall performance.